### DISCUSSION TITLE
Count action ending up as aggregate

### DISCUSSION THREAD

**Discussion by vuki656** [prisma/prisma#12129] [Q&A] [answered]
I created a middlewere that would override the default `count` and count only entities where `isDeleted: false` like so:

```ts
orm.$use(async (parameters, next) => {
    // Override native `count` and only count non soft deleted entities
    if (parameters.action === 'count') {
        parameters.args = {
            where: {
                isDeleted: false,
            },
        }
    }

    return next(parameters)
})
```

Then I call the `count` like so: 
```ts
const foundSectionsCount = await orm.section.count()
```

What I am expecting to happen is the middleware will be triggered since we are calling `count` and it is checking if the action is named `count`.

Instead, it outputs the `action` as `aggregate`.

Here is the `parameters` log like so:

```
action:  {
  args: { select: { _count: [Object] } },
  dataPath: [],
  runInTransaction: false,
  action: 'aggregate',
  model: 'Section'
}
```

Im assuming `count` is using `aggreate` under the hood but I think the action should persist to be called `count` if we are calling `count`. Especially since the type for `action` specifies `count` exists: 

```ts
export type PrismaAction =
  | 'findUnique'
  | 'findMany'
  | 'findFirst'
  | 'create'
  | 'createMany'
  | 'update'
  | 'updateMany'
  | 'upsert'
  | 'delete'
  | 'deleteMany'
  | 'executeRaw'
  | 'queryRaw'
  | 'aggregate'
  | '**count**'
  | 'runCommandRaw'
```

**Comment by austincrim** [comment, marked as answer]
Hey @vuki656 👋 ,

This looks like a bug! Would you mind opening a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with this info? Our engineering teams will pick it up from there.

Thanks for bringing it to our attention!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
